diff --git a/crt/s2n/crypto/s2n_fips.c b/crt/s2n/crypto/s2n_fips.c
index 13f9f77c0..b8af2e36e 100644
--- a/crt/s2n/crypto/s2n_fips.c
+++ b/crt/s2n/crypto/s2n_fips.c
@@ -57,15 +57,15 @@ int s2n_fips_init(void)
 {
     s2n_fips_mode_enabled = s2n_libcrypto_is_fips();
 
-    /* When using Openssl, ONLY 3.0 currently supports FIPS.
-     * openssl-1.0.2-fips is no longer supported.
-     * openssl >= 3.5 will likely have a FIPS 140-3 certificate instead of a
-     * FIPS 140-2 certificate, which will require additional review in order
-     * to properly integrate.
-     */
-#if defined(OPENSSL_FIPS) || S2N_OPENSSL_VERSION_AT_LEAST(3, 5, 0)
-    POSIX_ENSURE(!s2n_fips_mode_enabled, S2N_ERR_FIPS_MODE_UNSUPPORTED);
-#endif
+//     /* When using Openssl, ONLY 3.0 currently supports FIPS.
+//      * openssl-1.0.2-fips is no longer supported.
+//      * openssl >= 3.5 will likely have a FIPS 140-3 certificate instead of a
+//      * FIPS 140-2 certificate, which will require additional review in order
+//      * to properly integrate.
+//      */
+// #if defined(OPENSSL_FIPS) || S2N_OPENSSL_VERSION_AT_LEAST(3, 5, 0)
+//     POSIX_ENSURE(!s2n_fips_mode_enabled, S2N_ERR_FIPS_MODE_UNSUPPORTED);
+// #endif
 
     return S2N_SUCCESS;
 }
